### PR TITLE
DAOS-13886 test: Fix version functional test

### PR DESCRIPTION
**Pull request by knard-intel**
# Description

Follow up of the PR #12570 fixing non standard coding guidelines introuce in the functional test control/version.py

# Before requesting gatekeeper:

* [x] Two review approvals and any prior change requests have been resolved.
* [x] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [x] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [x] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [x] Any tests skipped by the ticket being addressed have been run and passed in the PR.

# Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by knard-intel** [comment]
CI test failure is a known issue:
- [DAOS-13931](https://daosio.atlassian.net/browse/DAOS-13931): issue [daos_test/suite.py:DaosCoreTest.test_daos_extend_simple - timeout after enumerate objects during extend](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/5/testReport/FTEST_daos_test/DaosCoreTest/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___20___daos_test_suite_py_DaosCoreTest_test_daos_extend_simple_run_agent_config_transport_config_daos_tests_args_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_bbfe/)

File: src/tests/ftest/control/version.py [hunk 1]
```diff
@@ -56,25 +36,21 @@ def test_version(self):
...
                 rpm_version = result[0]
                 break
         if not result:
-            msg = "RPM version could not be defined"
-            self.append_error(msg, stdout)
-            self.report_result()
+            report_errors(self, ["RPM version could not be defined"])
         self.log.info("RPM version = %s", rpm_version)
 
         # Remove configuration files
```
Comment (knard-intel) [inline src/tests/ftest/control/version.py:56]: @phender , as indicated in the first PR, I would like to keep these cleanup actions.
Indeed, the /etc/daos configuration files seems to not be properly cleaned between each tests and configuration files such as certificates are not cleaned.
```
22:22:59 INFO | Command: sudo find /etc/daos/certs -type f -delete -print
Results:
  wolf-105vm2: exit_status=0, interrupted=False:    /etc/daos/certs/server.crt
    /etc/daos/certs/clients/agent.crt
    /etc/daos/certs/server.key
    /etc/daos/certs/daosCA.crt
```

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/util/general_utils.py [hunk 2]
```diff
@@ -1332,7 +1346,7 @@ def report_errors(test, errors):
             len(errors), "" if len(errors) == 0 else "s"))
         test.fail(error_msg)
 
-    test.log.info("No errors detected.")
+    test.log.info("Test passed")
```
Comment (daltonbohning) [inline src/tests/ftest/util/general_utils.py:1349]: This function is used by other tests as well, and generally might be called multiple times. So "Test passed" isn't always correct. I.e. if the test continues to do more work and then calls report_errors again

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/general_utils.py (hunk 2 above)
Comment (knard-intel) [reply src/tests/ftest/util/general_utils.py]: Fixed with commit 6e503c582

**Review by daltonbohning — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Review by shimizukko — APPROVED** [review]

**Comment by knard-intel** [comment]
The only test failure is a known issue:
[DAOS-13931](https://daosio.atlassian.net/browse/DAOS-13931]: test failure [daos_test/suite.py:DaosCoreTest.test_daos_extend_simple - timeout after enumerate objects during extent](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/7/testReport/FTEST_daos_test/DaosCoreTest/Test_Hardware___Functional_Hardware_Medium_Verbs_Provider___20___daos_test_suite_py_DaosCoreTest_test_daos_extend_simple_run_agent_config_transport_config_daos_tests_args_daos_test_num_clients_pools_created_scalable_endpoint_stopped_ranks_test_name_dmg_hosts_pool_server_config_engines_0_storage_0_1_timeouts_fe90/)

**Comment by phender** [comment]
@knard-intel unfortunately the PR did not run the modified test: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/7/testReport/FTEST_control/

Please re-run the PR with the following commit pragmas to verify the changes:
```
Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: test_version
```

**Comment by phender** [comment]
The modified test did pass: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/7/artifact/Functional%20on%20EL%208/control/version.py/results.html

**Comment by phender** [comment]
> The modified test did pass: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/7/artifact/Functional%20on%20EL%208/control/version.py/results.html
> 
> For some reason this result is not included in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/7/testReport/FTEST_control/ despite no obvious issues in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/7/artifact/Functional%20on%20EL%208/control/version.py/results.xml

Ah, I figured it out.  The Test Result is recorded in https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-12615/6/testReport/FTEST_control/DAOSVersion/ - where the Functional on EL8 stage ran - and the artifacts in build 7 include those from 6 & 7.

**Pull request merged**